### PR TITLE
feat(trading): show oracle spec in index price stats

### LINKED ISSUE #5942: Tooltip on Index price to link to explorer for that oracle
## The Chore

What we need to achieve and who for

## Tasks

- [ ]
- [ ]

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #5942 

# Description ℹ️

Show Oracle spec in index price stats. 

# Demo 📺

image

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

File: libs/orders/src/lib/components/order-data-provider/order-data-provider.ts [hunk 1]
```diff
@@ -131,7 +131,8 @@ export const update = <T extends Omit<OrderFieldsFragment, 'market'> & Cursor>(
 ): T[] => {
   const updatedData = data ? [...data] : ([] as T[]);
   delta.forEach((orderUpdate) => {
-    const index = data?.findIndex((order) => order.id === orderUpdate.id) ?? -1;
+    const index =
+      data?.findIndex((order) => order?.id === orderUpdate.id) ?? -1;
```
Comment (mattrussell36) [inline libs/orders/src/lib/components/order-data-provider/order-data-provider.ts:135]: Why these changes the seem completely unrelated to the rest of this PR? Are the types incorrect here and order can be undefined?

Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-data-provider/order-data-provider.ts:135]: weirdly while I was testing i got some exceptions on stagnet1 and order was undefined

Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-data-provider/order-data-provider.ts:135]: image

Comment (MadalinaRaicu) [reply libs/orders/src/lib/components/order-data-provider/order-data-provider.ts:135]: added that "?." since it is an extra check that does not hurt and prevents runtime exceptions... but yeah, spotted that on staging only

**Review by daro-maj — APPROVED** [review]

File: libs/orders/src/lib/components/order-data-provider/order-data-provider.ts (hunk 1 above)
Comment (mattrussell36) [reply libs/orders/src/lib/components/order-data-provider/order-data-provider.ts:135]: Its very annoying to have to clutter the code with syntax to account for poorly typed objects. Its fine for now. @bglownia we probably want to take a look at this

**Review by mattrussell36 — APPROVED** [review]
